src/units.js

```diff
@@ -0,0 +1,74 @@
+/* eslint no-console:0 */
+
+/**
+ * This file does conversion between units.  In particular, it provides
+ * calculateSize to convert other units into ems.
+ */
+
+const ParseError = require("./ParseError");
+
+// This table gives the number of pts in one of each *absolute* TeX unit.
+// Thus, multiplying a length by this number converts the length from units
+// into pts.  Dividing the result by ptPerEm gives the number of ems
+// *assuming* a font size of ptPerEm (normal size, normal style).
+const ptPerUnit = {
+    // https://en.wikibooks.org/wiki/LaTeX/Lengths and
+    // https://tex.stackexchange.com/a/8263
+    "pt": 1,            // TeX point
+    "mm": 7227 / 2540,  // millimeter
+    "cm": 7227 / 254,   // centimeter
+    "in": 72.27,        // inch
+    "bp": 803 / 800,    // big (PostScript) points
+    "pc": 12,           // pica
+    "dd": 1238 / 1157,  // didôt
+    "cc": 14856 / 1157, // cîcero (12 didôt)
+    "nd": 685 / 642,    // new didôt
+    "nc": 1370 / 107,   // new cîcero (12 new didôt)
+    "sp": 1 / 65536,    // scaled point (TeX's internal smallest unit)
+    // https://tex.stackexchange.com/a/41371
+    "px": 803 / 800,    // \pdfpxdimen defaults to 1 bp in pdfTeX and LuaTeX
+};
+
+// This table gives the number of LaTeX ems in one of each *relative* TeX
+// unit (which depends on font size), except ex which is font-dependent.
+const emPerUnit = {
+    "em": 1,
+    "mu": 1 / 18,       // math unit
+};
+
+/**
+ * Determine whether the specified unit (either a string defining the unit
+ * or a "size" parse node containing a unit field) is valid.
+ */
+const validUnit = function(unit) {
+    if (unit.unit) {
+        unit = unit.unit;
+    }
+    return (unit in ptPerUnit || unit in emPerUnit || unit === "ex");
+};
+
+/**
+ * Convert a "size" parse node (with numeric "number" and string "unit" fields)
+ * into the equivalent number of LaTeX ems in the current font.  Note that
+ * this function does *not* scale according to the style, as the LaTeX em unit
+ * does not change when switching e.g. to scriptstyle.  But it uses the
+ * specified style's font to translate ex units into em units.
+ */
+const calculateSize = function(sizeValue, options) {
+    let x = sizeValue.number;
+    if (sizeValue.unit in emPerUnit) {
+        x *= emPerUnit[sizeValue.unit];
```

Comment:
I'd recommend making changes based on #755. `em` and `mu` should scale by the `quad` font metric; `em` should be additionally un-scaled if the current style is script/scriptscript.